api/notifications_and_subscriptions.rb

```diff
@@ -28,3 +28,26 @@
 delete "#{APIPREFIX}/users/:user_id/subscriptions" do |user_id|
   user.unsubscribe(source).to_hash.to_json
 end
+
+get "#{APIPREFIX}/subscriptions/:thread_id" do |thread_id|
+  page = (params['page'] || DEFAULT_PAGE).to_i
+  per_page = (params['per_page'] || DEFAULT_PER_PAGE).to_i
+
+  # Build a query hash based on the query parameters
+  query = {}
+  query[:subscriber_id] = params[:subscriber_id] if params[:subscriber_id]
+  query[:source_id] = thread_id
+  query[:source_type] = params[:source_type] if params[:source_type]
```

Comment:
I think we can set this to `CommentThread`, as this endpoint is specific to threads only!
If we want this to be interchangable, i suggest we change the `thread_id` param in the endpoint to something like `commentable_id`